tests/layering/helpers_test.go

```diff
@@ -0,0 +1,611 @@
+	t.Logf("using %s as the base image", baseImageBuildArg)
+
+	// Create a new build
+	buildConfig := &buildv1.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: buildName,
+		},
+		Spec: buildv1.BuildSpec{
+			CommonSpec: buildv1.CommonSpec{
+				Source: buildv1.BuildSource{
+					Type: "Git",
+					Git: &buildv1.GitBuildSource{
+						URI: "https://github.com/coreos/fcos-derivation-example",
+						Ref: "rhcos",
+					},
+				},
+				Strategy: buildv1.BuildStrategy{
+					DockerStrategy: &buildv1.DockerBuildStrategy{
+						BuildArgs: []corev1.EnvVar{
+							{
+								Name:  "RHEL_COREOS_IMAGE",
+								Value: baseImageBuildArg,
+							},
+						},
+					},
+				},
+				Output: buildv1.BuildOutput{
+					To: &v1.ObjectReference{
+						Kind: "ImageStreamTag",
+						Name: imageStreamName + ":latest",
+					},
+				},
+			},
+		},
+	}
+
+	// Start our new build
+	_, err = cs.BuildV1Interface.Builds(mcoNamespace).Create(context.TODO(), buildConfig, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("could not create build: %w", err)
+	}
+
+	t.Log("base image derivation build started")
+
+	build, err := cs.BuildV1Interface.Builds(mcoNamespace).Get(context.TODO(), buildConfig.Name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("could not get build %s: %w", build.Name, err)
+	}
+
+	// Wait for the build to complete
+	return waitForBuildToRun(t, cs, build)
+}
+
+// Puts the pull secret for the MCD service account on the underlying node so rpm-ostree can use it.
+// get ImagePullSecret for the MCD service account and save to authfilePath on the node
+// eventually we should use rest.InClusterConfig() instead of cs with kubeadmin
+func putPullSecretOnNode(t *testing.T, cs *framework.ClientSet, node *corev1.Node) func() {
+	t.Logf("placing image pull secret on node")
+
+	// Get the MCD service account
+	mcdServiceAccount, err := cs.ServiceAccounts(mcoNamespace).Get(context.TODO(), "machine-config-daemon", metav1.GetOptions{})
+	require.Nil(t, err)
+	require.Equal(t, 1, len(mcdServiceAccount.ImagePullSecrets))
+
+	// Get the image pull secret attached to the service account
+	imagePullSecret, err := cs.Secrets(mcoNamespace).Get(context.TODO(), mcdServiceAccount.ImagePullSecrets[0].Name, metav1.GetOptions{})
+	require.Nil(t, err)
+
+	// Extract the dockerconfig from the pull secret
+	dockerConfigData := imagePullSecret.Data[corev1.DockerConfigKey]
+	var dockerConfig credentialprovider.DockerConfig
+	err = json.Unmarshal(dockerConfigData, &dockerConfig)
+	require.Nil(t, err)
+	dockerConfigJSON := credentialprovider.DockerConfigJSON{
+		Auths: dockerConfig,
+	}
+	authfileData, err := json.Marshal(dockerConfigJSON)
+	require.Nil(t, err)
+
+	require.Nil(t, writeFileToNode(t, cs, node, authfileData, authfilePath))
+
+	return func() {
+		// Clean up the pull secret from the node if the node has not rebooted and we do not want to delete it.
+		t.Logf("cleaning up pull secret from node")
```

Comment:
logf->log or alternatively add "node: %s, node.name" (or something)

Reply:
I made that mistake a lot; sorry! I'll get them all fixed and make a note to tune my linter.